lib/schema.js

```diff
@@ -818,15 +852,71 @@ Schema.prototype.add = function add(obj, prefix) {
         }
       }
     }
+
+    if (val.instanceOfSchema && val.encryptionType() != null) {
+      // schema.add({ field: <instance of encrypted schema> })
+      if (this.encryptionType() != val.encryptionType()) {
```

Comment:
Mongoose document arrays get implicitly created schemas, so `docArr: [{ name: String }]` implicitly creates a schema with `{ name: String }`. We pass certain schema options to implicitly created schemas, like `typeKey`. It might be worthwhile to make sure encryption type gets passed down as well.

Reply:
Arrays are encrypted as an array, not by individual element.  So the logic here only creates a schemaType of type "array" - the contents of the array don't matter.  So encryptionType for the implicitly generated schemas shouldn't matter.

That being said, I'm happy to ensure that the encryption type gets propagated to sub-schemas if you'd like

Reply:
Ok, if encryption type doesn't matter for implicitly created schemas then that should be fine